library/bignum_core.c

```diff
@@ -434,4 +434,20 @@ void mbedtls_mpi_core_montmul( mbedtls_mpi_uint *X,
     mbedtls_ct_mpi_uint_cond_assign( AN_limbs, X, T, (unsigned char) ( carry ^ borrow ) );
 }
 
+int mbedtls_mpi_get_montgomery_constant_unsafe( mbedtls_mpi *X,
+                                                mbedtls_mpi const *N )
+{
+    int ret = MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED;
+
+    if ( X == NULL || N == NULL ) goto cleanup;
```

Comment:
Is `MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED` the correct return code for `NULL` inputs?

Reply:
No: this is a “can't happen” error. A function should never have this error documented with conditions that can happen. If NULL is a documented possible argument then `BAD_INPUT_DATA` is the correct error code. But why does this function even allow null arguments? I don't see why any bignum function would allow a null pointer where an MPI (in any form) is expected.

Reply:
So should this line just be removed?

Reply:
This check was meant to avoid a segfault in the occasion that a null pointer is passed over as an argument. I can update the return code to `MBEDTLS_ERR_MPI_BAD_INPUT_DATA` or remove the line. I am included to do the former.

This function is already planned to be rewritten in the roadmap anyway, since it utilised the unsafe modular reduction.

Reply:
Yes (and any similar line in `bignum*.c`).